src/components/index/IndexComponent.tsx

```diff
@@ -7,11 +8,68 @@ import {
   Body,
   TableRow,
   Logo,
+  FilterButtonRow,
+  FilterButton,
+  FilterWrapper,
+  FilterNumbers,
 } from "./Index.styled";
 import RequestsTable from "./RequestsTable";
 import ooLogo from "assets/uma-oo-logo-redcirclebg.svg";
+import useClient from "hooks/useOracleClient";
+import useReader from "hooks/useOracleReader";
+import { RequestState } from "constants/blockchain";
+import { RequestIndex } from "@uma/sdk/dist/types/oracle/types/state";
+
+enum Filter {
+  DEFAULT,
+  REQUESTS,
+  PROPOSED,
+  DISPUTED,
+  ANSWERED,
+}
 
 const Index = () => {
+  const [filter, setFilter] = useState<Filter>(Filter.DEFAULT);
+  const { state } = useClient();
+  const { descendingRequests } = useReader(state);
+  const [filteredRequests, setFilteredRequests] = useState(descendingRequests);
+  const [checked, setChecked] = useState<boolean>(false);
+  const [numAll, setNumAll] = useState(0);
+  const [numProposed, setNumProposed] = useState(0);
+  const [numRequested, setNumRequested] = useState(0);
+  const [numDisputed, setNumDisputed] = useState(0);
+  const [numAnswered, setNumAnswered] = useState(0);
+
+  useEffect(() => {
+    // Default state, IE: filter = Filter.DEFAULT
+    let filterFunc = ALL_FILTER;
+
+    if (!checked) {
+      if (filter === Filter.PROPOSED) {
+        filterFunc = PROPOSED_FILTER;
+      }
+      if (filter === Filter.REQUESTS) {
+        filterFunc = REQUEST_FILTER;
+      }
+      if (filter === Filter.DISPUTED) {
+        filterFunc = DISPUTE_FILTER;
+      }
+    } else {
+      filterFunc = ANSWERED_FILTER;
+    }
+
+    const fr = descendingRequests.filter(filterFunc);
+    setFilteredRequests(fr);
+  }, [filter, descendingRequests, checked]);
+
+  useEffect(() => {
+    setNumAll(descendingRequests.filter(ALL_FILTER).length);
+    setNumRequested(descendingRequests.filter(REQUEST_FILTER).length);
+    setNumProposed(descendingRequests.filter(PROPOSED_FILTER).length);
+    setNumDisputed(descendingRequests.filter(DISPUTE_FILTER).length);
+    setNumAnswered(descendingRequests.filter(ANSWERED_FILTER).length);
```

Comment:
u could probably do this in a single loop, rather than iterating 5 times over the list of requests, but i dont think the performance is that bad at the moment

Reply:
What were you thinking?

Reply:
Use reduce on the requests array, and init it with 5 counters, then increment each counter based on the requests in the array. Then set the states based on the final count of the reducer. Imo it would be easy to have a single set state that took an object of each count then

Reply:
heres an example

```js
const filteredRequests = descendingRequests.reduce((result,request)=>{
  if(ALL_FILTEr(request)){
    result.all.push(request)
  }
  if(REQUEST_FILTER(request)){
     result.requested.push(request)
  }
  // ...etc
  return result
},{
// this is the initial object
all:[],
requested:[],
proposed:[],
disputed:[],
answered:[],
})

setNumAll(filteredRequests.all.length)
// ...etc, or just save the raw arrays so you can just switch them in base on the filter
```

Reply:
Implemented.